test/hooks/test-trace-grpc.js

```diff
@@ -199,69 +217,125 @@ Object.keys(versions).forEach(function(version) {
       });
     });
 
+    it('should trace errors for unary requests', function(done) {
+      common.runInTransaction(function(endTransaction) {
+        client.testUnary({n: 13412}, function(err, result) {
+          endTransaction();
+          assert(err);
+          var trace = common.getMatchingSpan(grpcPredicate);
+          assert(trace);
+          assert.strictEqual(trace.labels.argument, '{"n":13412}');
+          assert.strictEqual(trace.labels.error, 'test');
+          done();
+        });
+      });
+    });
+
+    it('should trace errors for client requests', function(done) {
+      common.runInTransaction(function(endTransaction) {
+        var stream = client.testClientStream(function(err, result) {
+          endTransaction();
+          assert(err);
+          var trace = common.getMatchingSpan(grpcPredicate);
+          assert(trace);
+          assert.strictEqual(trace.labels.error, 'test');
+          done();
+        });
+        stream.write({n: 13412});
+        stream.end();
+      });
+    });
+
+    it('should trace metadata for server requests', function(done) {
+      common.runInTransaction(function(endTransaction) {
+        var stream = client.testServerStream({n: 42}, metadata);
+        stream.on('data', function(data) {});
+        stream.on('status', function(status) {
+          endTransaction();
+          assert.strictEqual(status.code, grpc.status.OK);
```

Comment:
Yes, we can listen for the 'status' event. I've added the test for that to the standard server stream request above (this test is for metadata).